test/tm-cases/toc_2.toc_html

```diff
@@ -6,8 +6,10 @@
     <ul>
       <li><a href="#beef">Beef</a>
       <ul>
-        <li><a href="#steak">Steak</a></li>
-        <li><a href="#burgers">Burgers</a></li>
+        <ul>
+          <li><a href="#steak">Steak</a></li>
+          <li><a href="#burgers">Burgers</a></li>
+        </ul></li>
```

Comment:
Is this the orphaned li you mentioned? Is it present in master right now? Also wondering why we now have double nested ul in here.

Reply:
Yup, that's the one. It probably doesn't exist in master right now, but I haven't taken a close look.

The double nested <ul> is the result of this PR. It allows for someone to place an <h5> after an <h3> and it indents twice in the TOC. This is happening in the toc_2 test case:

```### Beef

##### Steak

##### Burgers```

Reply:
Okay. I'll take a look at the code. Definitely not ideal to have the malformed html. The double ul doesn't seem necessary either, from a markup perspective. Seems the purpose of the PR should be able to be accomplished without an unnecessary ul. I'll dive in soon

Reply:
Personally, I think the nested `<ul>` is the right way to go. It seems odd, but it seems to follow good flow in the HTML where it relies on list level to determine indentation. It also means that markdown like this:

```
# Level 1

### Level 3

### Level 3

# Level 1
```

will be handled similarly to markdown like this: 

```
# Level 1

### Level 3

### Level 3

## Level 2

# Level 1
```

I'll also mess around with the code a bit and see if I can reorganize the `</li>` tags to be a bit clearer.